### PR TITLE
[TEST] Add Tests and Mocks for domain/models/.* in Backend

### PR DESCRIPTION
**Pull request by David-Xia0**
Created tests for domain/models/Game.js and domain/models/SocketSession.js.

Required to create mock for mongo/models/session.js

Note: Game.js requires socket.io client and other socket functions

### CODE REVIEW

File: backend/src/domain/models/Game.js [hunk 1]
```diff
@@ -78,6 +80,7 @@ export class Game {
    * This emits an end_game event to all users in the session
    */
   async endGame() {
+    // delay to let db sync, thne tell suers game has ended
```
Comment (afei088) [inline backend/src/domain/models/Game.js:83]: I think there are a couple of typos in this comment. Specifically, "thne" and "suers". Would be nice if you could change those to "then" and "users". :)

File: backend/src/domain/models/Game.js [hunk 2]
```diff
@@ -25,6 +25,7 @@ export class Game {
    * 0 represents the start of the game (calling nextRound)
    */
   async startCountdown() {
+    // let all users in game start countdown
```
Comment (afei088) [inline backend/src/domain/models/Game.js:28]: Maybe start these comments with a capital letter, to keep it consistent with the rest of the project. :)

File: backend/src/domain/models/Game.js (hunk 1 above)
Comment (Tianrens) [reply backend/src/domain/models/Game.js:83]: Yea, would also be good if punctuation was consistent. Like sentenses starting with a captical letter and ending with (.).

File: backend/src/domain/models/Game.js [hunk 3]
```diff
@@ -39,7 +40,7 @@ export class Game {
   }
 
   /**
-   *
+   * sleep function
```
Comment (afei088) [inline backend/src/domain/models/Game.js:43]: Same with these comments. That would be cool.

File: backend/src/domain/models/Game.js (hunk 1 above)
Comment (David-Xia0) [reply backend/src/domain/models/Game.js:83]: Oo, thank you was in a bit of a rush I can fix that later today

File: backend/src/domain/models/Game.js [hunk 4]
```diff
@@ -60,6 +61,7 @@ export class Game {
 
 
     for (let i=0; i < this.swipeDeck.length; i++) {
+      // notifies users of round change
```
Comment (afei088) [inline backend/src/domain/models/Game.js:64]: And also this. Same goes for the remaining of the PR. Not to important, but would just be nice to keep things cleaner.

File: backend/src/domain/models/Game.js (hunk 2 above)
Comment (David-Xia0) [reply backend/src/domain/models/Game.js:28]: ignore the emoji, it wont go away

**Review by afei088 — DISMISSED** [review]
Just some minor comment stylistic tweaks to be made, otherwise LGTM!

**Review by Tianrens — APPROVED** [review]
LGTM

**Review by afei088 — APPROVED** [review]
Awesome work!

**Pull request merged**